framework/src/base/MooseInit.C

```diff
@@ -21,6 +21,22 @@
 #include <omp.h>
 #endif
 
+#include <unistd.h>
+#include <signal.h>
+
+void
+my_sig_handler(int signum)
+{
+  Moose::autosave_flag = signum;
```

Comment:
Just thinking about the name some more - if two different signals arrive, we'd again only capture the higher numbered one of the two. I'm not suggesting changing anything about this implementation, just thinking about weird ways this could misbehave.

Reply:
It should just capture the most recent signal, not the highest numbered one. It also can only capture SIGUSR1 for now.

Reply:
Yes, I understand the design and it's fine for now since we only have one. There are ways we could capture multiple at the same time with a simple "bitset" variable. No need to change anything now.